test-infra/src/main/java/com/datastax/oss/driver/api/testinfra/simulacron/SimulacronRule.java

```diff
@@ -24,14 +24,22 @@
 import com.datastax.oss.simulacron.server.Inet4Resolver;
 import com.datastax.oss.simulacron.server.Server;
 import java.net.InetSocketAddress;
+import java.net.SocketAddress;
 import java.util.Set;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.stream.Collectors;
 
 public class SimulacronRule extends CassandraResourceRule {
   // TODO perhaps share server some other way
+  // TODO: Temporarily do not release addresses to ensure IPs are always ordered
```

Comment:
I suspect it could be a problem in tests that use `SortingLoadBalancingPolicy` and then using `simulacronCluster.node(X)` with the expectation that nodes are ordered within a simulacron cluster by IP address, which may not be the case if simulacron uses reuses IP addresses.  Something i need to look into more since I thought this was a cause of some specex test errors, but apparently isn't.

Reply:
It looks like the source of the problem was that `buildSessionWithProfile` did not use `SortingLoadBalancingPolicy`, so order was never enforced, and somehow this didn't happen to be a problem most of the time.  I found it odd that only profile-based tests were failing, this explains it! (342b695)